gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractNotarizationTask.kt

```diff
@@ -5,24 +5,78 @@
 
 package org.jetbrains.compose.desktop.application.tasks
 
-import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Nested
-import org.gradle.api.tasks.Optional
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.tasks.*
 import org.jetbrains.compose.desktop.application.dsl.MacOSNotarizationSettings
-import org.jetbrains.compose.internal.utils.nullableProperty
+import org.jetbrains.compose.desktop.application.dsl.TargetFormat
+import org.jetbrains.compose.desktop.application.internal.files.checkExistingFile
+import org.jetbrains.compose.desktop.application.internal.files.findOutputFileOrDir
+import org.jetbrains.compose.desktop.application.internal.validation.ValidatedMacOSNotarizationSettings
 import org.jetbrains.compose.desktop.application.internal.validation.validate
 import org.jetbrains.compose.desktop.tasks.AbstractComposeDesktopTask
+import org.jetbrains.compose.internal.utils.MacUtils
+import org.jetbrains.compose.internal.utils.ioFile
+import java.io.File
+import javax.inject.Inject
 
-abstract class AbstractNotarizationTask : AbstractComposeDesktopTask() {
+abstract class AbstractNotarizationTask @Inject constructor(
     @get:Input
-    @get:Optional
-    internal val nonValidatedBundleID: Property<String?> = objects.nullableProperty()
+    val targetFormat: TargetFormat
+) : AbstractComposeDesktopTask() {
 
     @get:Nested
     @get:Optional
     internal var nonValidatedNotarizationSettings: MacOSNotarizationSettings? = null
 
-    internal fun validateNotarization() =
-        nonValidatedNotarizationSettings.validate(nonValidatedBundleID)
-}
\ No newline at end of file
+    @get:InputDirectory
+    val inputDir: DirectoryProperty = objects.directoryProperty()
+
+    init {
+        check(targetFormat != TargetFormat.AppImage) { "${TargetFormat.AppImage} cannot be notarized!" }
+    }
+
+    @TaskAction
+    fun run() {
+        val notarization = nonValidatedNotarizationSettings.validate()
+        val packageFile = findOutputFileOrDir(inputDir.ioFile, targetFormat).checkExistingFile()
+
+        notarize(notarization, packageFile)
+        staple(packageFile)
+    }
+
+    private fun notarize(
+        notarization: ValidatedMacOSNotarizationSettings,
+        packageFile: File
+    ) {
+        logger.info("Uploading '${packageFile.name}' for notarization")
+
+        val args = arrayListOf<String>().apply {
+            add("notarytool")
+            add("submit")
+            add("notarytool")
+            add("--wait")
+            add("--apple-id")
+            addAll(listOf(
+                "notarytool",
+                "submit",
+                "--wait",
+                "--apple-id",
+                notarization.appleID
+            ))
+            if (notarization.teamID != null) {
+                add("--team-id")
+                add(notarization.teamID)
+            }
+            add(packageFile.absolutePath)
+        }
```

Comment:
This block of code is hard to read.
As I understand we need to run these commands:
xcrun notarytool submit
xcrun notarytool --wait --apple-id
xcrun notarytool submit --wait --apple-id <APPLE_ID> --team-id <TEAM_ID> <PATH>

Is it correct?

Reply:
Good catch, I added some extra arguments accidentely, when moving the code around :)